src/main/java/Controller/LoginController.java

```diff
@@ -0,0 +1,27 @@
+package Controller;
+
+import InputBoundary.loginInputBoundary;
+import InputData.loginInputData;
+
+
+/**
+ * Controller for login and register use cases.
+ */
+public class LoginController {
+    private final loginInputBoundary loginInteractor;
+    private final loginInputData loginInputData;
+    /**
+     * Constructor.
+     * @param loginInteractor use case interactor for user logging in
+     */
+    public LoginController(loginInputBoundary loginInteractor, String username, String password, boolean isArtist) {
```

Comment:
The initializer should be passed a presenter, which is then passed to the interactor.

The controller initializer should also construct the loginInteractor (Just for the sake of consistency, as Sarah previously mentioned) rather than be passed it.

The ViewModel would be the one creating the InputData object, passed directly to the login() rather than to the initializer -- I imagine creating new instances of LoginController for each login attempt would be problematic.